app.py

```diff
@@ -141,15 +153,29 @@ def save_file_to_temp(file_obj):
     # Display data from session state
     if 'school_center' in st.session_state.calculated_data:
         df_school_center = pd.read_csv(st.session_state.calculated_data['school_center'], sep="\t")
-        allowed_filter_types = ['scode', 'school', 'cscode', 'center']
+        allowed_filter_types = ['school', 'center']
         st.session_state.filter_type = tab1.radio("Choose a filter type:", allowed_filter_types)
 
         # Display an input field based on the selected filter type
         if st.session_state.filter_type:
-            st.session_state.filter_value = tab1.selectbox(f"Select a value for {st.session_state.filter_type}:", df_school_center[st.session_state.filter_type].unique())
-        # Filter the DataFrame based on the selected filter type and value
+         if st.session_state.filter_type == 'school':
+          # Create filter options with school name and code
+          filter_options = [f"{name} | [{code}]" for name, code in zip(df_school_center['school'], df_school_center['scode'])]
+
+         elif st.session_state.filter_type == 'center':
+          # Create filter options with center name and code
+          filter_options = [f"{name} | [{code}]" for name, code in zip(df_school_center['center'], df_school_center['cscode'])]
+
+         # Display a selectbox for selection
+         st.session_state.filter_value = tab1.selectbox(f"Select a value for {st.session_state.filter_type}:", filter_options)
+
+         # Split the selected value to extract name and code
+         name, code = st.session_state.filter_value.split(' | ')
+
+         # Filter the DataFrame based on the selected type and value
+         filtered_df = filter_data(df_school_center, st.session_state.filter_type, name)
+
         if st.session_state.filter_value:
-            filtered_df = filter_data(df_school_center, st.session_state.filter_type, st.session_state.filter_value)
             tab1.dataframe(filtered_df)
```

Comment:
```suggestion
            tab1.dataframe(filtered_df, hide_index=True)
```
This will hide the index. Feel free to add for each `dataframe`